source/image-handler/image-request.ts

```diff
@@ -245,25 +245,26 @@ export class ImageRequest {
    * @returns The edits to be made to the original image.
    */
   public parseImageEdits(event: ImageHandlerEvent, requestType: RequestTypes): ImageEdits {
-    if (requestType === RequestTypes.DEFAULT) {
-      const decoded = this.decodeRequest(event);
-      return decoded.edits;
-    } else if (requestType === RequestTypes.THUMBOR) {
-      const thumborMapping = new ThumborMapper();
-      return thumborMapping.mapPathToEdits(event.path);
-    } else if (requestType === RequestTypes.SEMANTIC) {
-      const semanticMapping = new SemanticMapper();
-      return semanticMapping.mapPathToEdits(event);
-    } else if (requestType === RequestTypes.CUSTOM) {
-      const thumborMapping = new ThumborMapper();
-      const parsedPath = thumborMapping.parseCustomPath(event.path);
-      return thumborMapping.mapPathToEdits(parsedPath);
-    } else {
-      throw new ImageHandlerError(
-        StatusCodes.BAD_REQUEST,
-        "ImageEdits::CannotParseEdits",
-        "The edits you provided could not be parsed. Please check the syntax of your request and refer to the documentation for additional guidance."
-      );
+    const thumborMapping = new ThumborMapper();
+    const semanticMapping = new SemanticMapper();
```

Comment:
If you define them upfront and not within switch case you create both mappers, this wile only one of them is going to be used.
If you do that, at least use a lazy loading lambda function.